MineSweeper.py

```diff
@@ -0,0 +1,256 @@
+    
+    row = []
+
+    for i in range(self.__length):
+      for j in range(self.__width):
+        new_tile = Tile(i*BOX_SIZE, j*BOX_SIZE, BOX_SIZE)
+        row.append(new_tile)
+
+      self.__tiles.append(row)
+      row = []
+
+
+  def draw(self, panel):
+
+    panel.delete(ALL)
+
+    marked_mines = 0
+
+    for row in self.__tiles:
+      for tile in row:
+        tile.draw(panel)
+      
+        if(tile.marked_mine):
+          marked_mines += 1
+
+    mines = NUMBER_MINES - marked_mines
+
+    bottom_text = 'Mines left: ' + repr(mines)
+    text_fill = 'black'
+
+    panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 20, text  = bottom_text, fill = text_fill)
+
+
+  # assign mines to random locations across the board. 
+  def assign_mines(self):
+
+    mines_to_assign = NUMBER_MINES
+
+    while(mines_to_assign > 0):
+      x_index = random.randint(0, len(self.__tiles)-1)
+      y_index = random.randint(0, len(self.__tiles)-1)
+
+
+      if(self.__tiles[x_index][y_index].is_mine == False):
+        self.__tiles[x_index][y_index].is_mine = True
+        mines_to_assign -= 1
+
+        for i in range(-1, 2):
+          for j in range(-1, 2):
+            try:
+              if((x_index + i >= 0) and (y_index + j >= 0)):
+                self.__tiles[x_index + i][y_index + j].value += 1
+            except IndexError:
+              pass
+
+
+  def show_neighbors(self, center_x, center_y):
+  
+    for i in range(-1, 2):
+      for j in range(-1, 2):
+        try:
+          if(center_x + i >= 0) and (center_y + j >= 0):
+            if(self.__tiles[center_x + i][center_y+ j].is_clicked == False):
+              self.__tiles[center_x + i][center_y + j].show_self()
+
+              if(self.__tiles[center_x + i][center_y+ j].value == 0):
+                self.show_neighbors(center_x + i, center_y + j)
+
+        except IndexError:  
+          pass
+        
+
+  # reveals the tile that was clicked
+  def left_click_event(self, event):
+ 
+    print("HELLA LEFT CLIX")
+
+    mouse_click = []
+
+    mouse_click.append(event.x)
+    mouse_click.append(event.y)
+
+    box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+    box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+    try:
+      self.__tiles[box_index_x][box_index_y].show_self()
+
+      if(self.__tiles[box_index_x][box_index_y].value == 0):
+        self.show_neighbors(box_index_x, box_index_y)
+
+      if(self.__tiles[box_index_x][box_index_y].is_mine):
+        game.game_over()
+
+    except IndexError:
+      pass
+
+    
+    self.draw(game.panel)
+
+  # Marks the clicked square as a mine
+  def right_click_event(self, event):
+
+    print("HELLA RIGHT CLIX")
+
+    mouse_click = []
+
+    mouse_click.append(event.x)
+    mouse_click.append(event.y)
+
+    box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+    box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+    try:
+      board.__tiles[box_index_x][box_index_y].marked_mine = True
+
+    except IndexError:
+      pass
```

Comment:
I believe it's just like a skip. In this context, we want to iterate over tiles surrounding a tile, but that runs the risk of throwing an Index Error if we try to look at a tile outside of the board area. So, I'm saying if that happens, then we don't care, just keep on goin!